cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -567,6 +567,54 @@ const char* castVARCHAR_utf8_int64(gdv_int64 context, const char* data,
   return data;
 }
 
+// Split a string into substrings, then return a particular one
+FORCE_INLINE
+const char* split_utf8_utf8_int32(gdv_int64 context, const char* str, gdv_int32 str_len,
+                                  const char* delimiter, gdv_int32 delimiter_len,
+                                  gdv_int32 ret_index, gdv_int32* out_len) {
+  if (ret_index < 1) {
+    gdv_fn_context_set_error_msg(context, "Substring index must be greater than 0");
```

Comment:
does this match a specific sql function?  otherwise I would expect 0 based indexing?